src/lib.rs

```diff
@@ -89,38 +91,64 @@ impl Contract {
         }
     }
 
-    pub fn claim(&mut self) -> PromiseOrValue<WrappedBalance> {
+    pub fn claim(
+        &mut self,
+        amounts: Option<Vec<(LockupIndex, WrappedBalance)>>,
```

Comment:
It doesn't solve one of the use-cases. E.g. I don't want to pass `WrappedBalance`, but want to pass lockupIndex. Does it make sense to have switch to `amounts: Option<Vec<(LockupIndex, Option<WrappedBalance>)>>`?